src/zepben/evolve/streaming/data/current_state_event.py

```diff
@@ -0,0 +1,98 @@
+#  Copyright 2024 Zeppelin Bend Pty Ltd
+#  This Source Code Form is subject to the terms of the Mozilla Public
+#  License, v. 2.0. If a copy of the MPL was not distributed with this
+#  file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+from datetime import datetime
+from enum import Enum
+
+from zepben.protobuf.ns.data.change_events_pb2 import CurrentStateEvent as PBCurrentStateEvent, SwitchStateEvent as PBSwitchStateEvent
+
+from zepben.evolve.model.cim.iec61970.base.core.phase_code import PhaseCode, phase_code_by_id
+from zepben.evolve.util import datetime_to_timestamp
+
+
+@dataclass
+class CurrentStateEvent(ABC):
```

Comment:
todo: All your new classes (not tests) should be exported in an `__all__` collection and imported into the top level module so they can be imported directly from `zepben.evolve` from outside this codebase.

To make sure this is working, you should update all your test imports to pull your new classes in from `zepben.evolve` instead of their explicit packages.